executor/resources/sampler.go

```diff
@@ -31,8 +31,9 @@ type Sub[T WithTimestamp] struct {
 
 func (s *Sub[T]) Close(captureLast bool) ([]T, error) {
 	s.sampler.mu.Lock()
+	defer s.sampler.mu.Unlock()
```

Comment:
is it ok to hold on to this lock while `v, err := s.sampler.callback(time.Now())` is called? or does it need to be more granular?

Reply:
No, we should not hold the lock during callback. If the `Sub` is generic then callback could take a long time.

Reply:
that makes sense, it's been removed.